src/Symfony/Bundle/FrameworkBundle/CacheWarmer/RouterCacheWarmer.php

```diff
@@ -43,6 +43,8 @@ public function warmUp($cacheDir)
     {
         if ($this->router instanceof WarmableInterface) {
             $this->router->warmUp($cacheDir);
+        } else {
+            trigger_error(sprintf('%s is not a instanceof %s, no warmUp has been triggered.', get_class($this->router), WarmableInterface::class), \E_USER_NOTICE);
```

Comment:
IMHO we could trigger a deprecation now and throw a LogicException at the end.

Reply:
Ddo we ever trigger a notice in the code base? This is suspicious to me :)

Reply:
We trigger warning but not notices you are right, I will change that with @chalasr's suggestion

Reply:
Since we are triggering a deprecation, we should go with master ?

Reply:
Yes, triggering/throwing does not fix a bug, it only enhances DX. This is for master in any case IMO

Reply:
That's for 4.1 IMHO.

Reply:
Rebase and throw added.